D/Diamond/build_tarballs.jl

```diff
@@ -0,0 +1,45 @@
+using BinaryBuilder, Pkg
+
+name = "Diamond"
+version = v"2.1.6"
+
+# url = "https://github.com/bbuchfink/diamond"
+# description = "Accelerated BLAST-compatible local sequence aligner"
+
+# TODO
+# - WITH_ZSTD (fails to find the library)
```

Comment:
Uh?  https://buildkite.com/julialang/yggdrasil/builds/1583#0186f998-0be8-4174-96f7-f84ab2f847b5/637-796

Reply:
The log refers to ZLIB. With `Zstd_jll` as a dependency and `cmake -DWITH_ZSTD=1`, cmake can't find the zstd library.

Reply:
> The log refers to zlib. With Zstd_jll as a dependency and cmake -DWITH_ZSTD=1, cmake can't find the zstd library.

Sorry, right, I confused them

> Would be thankful for an idea how to fix this.

I'd need to know what the problem is, to make any guess.

Reply:
For the record, the problem was that diamond for some reasons expects zstd to be only a static library: https://github.com/bbuchfink/diamond/blob/0540f86d3d5965cf9a6ef8871068999ef61211dc/CMakeLists.txt#L470 :shrug:

Reply:
Many thanks for the help! That zstd is expected to be a static library seems to be quite common with cmake-based projects, I think some other packages that I made for Yggdrasil also might be able to use this fix.